### PR TITLE
Imosiychuk exceptions

### PR DESCRIPTION
**Pull request by Ivanko-Mosiychuk**

### CODE REVIEW

File: src/main/java/com/hotel/HotelAccount.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+        for (Room room : rooms) {
+            if (room.getSize() > size) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    //Перевіряємо чи в списку кімнат всі відповідають критерію
+    public boolean isAllRoomsPerFriendly(List<Room> rooms) {
```
Comment (nmelnyk-corevalue) [inline src/main/java/com/hotel/HotelAccount.java]: areAllRooms...

File: src/main/java/com/hotel/user/Admin.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        return null;
+    }
+
+    public void checkBookingRequest(Customer customer) {
+        if (customer.getAge() > 20 & !customer.getRoom().isBooking()) bookingRoom(customer.getRoom(), customer);
+    }
+
+
+
+    private void bookingRoom(Room room, Customer customer) {
```
Comment (nmelnyk-corevalue) [inline src/main/java/com/hotel/user/Admin.java]: book(Room room, Customer customer)

File: src/main/java/com/hotel/user/User.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            }
+            System.out.printf("%-12s %-12s %-12s %-12s %-9s %n",
+                    room.getClass().getSimpleName(), room.getNumber(), room.getSize(), room.getPrice(), s);
+        }
+        System.out.println("---------------------------------------------------------------");
+        System.out.println();
+
+    }
+
+    public void showCheckingResult(boolean result) {
```
Comment (nmelnyk-corevalue) [inline src/main/java/com/hotel/user/User.java]: boolean result -> convention 
isSuccessfull / isUnsuccessfull

File: src/main/java/com/hotel/user/Admin.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        return password;
+    }
+
+    public void setPassword(String password) {
+        this.password = password;
+    }
+
+    @Override
+    public Room chooseRoom(List<Room> rooms) {
+        return null;
```
Comment (nmelnyk-corevalue) [inline src/main/java/com/hotel/user/Admin.java]: null -> is not the best way

File: src/main/java/com/hotel/room/SingleRoom.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
+package com.hotel.room;
+
+import com.hotel.user.Customer;
+
+public class SingleRoom extends Room {
+
+    private Customer customer;
+
+    public SingleRoom(int number, double price, double size, boolean petFriendly) {
```
Comment (nmelnyk-corevalue) [inline src/main/java/com/hotel/room/SingleRoom.java]: boolean petFriendly -> isPetFriendly

File: src/main/java/com/hotel/room/SharedRoom.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    public int getNumberOfFreePlaces() {
+        return numberOfPlaces - numberOfBookingPlaces;
+    }
+
+    public void addNewBooking(Customer customer){
+        if (!isBooking()) {
+            guests.add(customer);
+            numberOfBookingPlaces++;
+        } else {
+            throw new NotFoundFreePlaceException("В загальній кімнаті №"+getNumber()+" вільні місця відсутні");
```
Comment (nmelnyk-corevalue) [inline src/main/java/com/hotel/room/SharedRoom.java:38]: great!

**Review by nmelnyk-corevalue — APPROVED** [review]
Fix the comments and merge please

**Comment by nmelnyk-corevalue** [comment]
This branch has conflicts that must be resolved

File: src/main/java/com/hotel/HotelAccount.java (hunk 1 above)
Comment (Ivanko-Mosiychuk) [reply src/main/java/com/hotel/HotelAccount.java]: Changed it.

File: src/main/java/com/hotel/user/Admin.java (hunk 2 above)
Comment (Ivanko-Mosiychuk) [reply src/main/java/com/hotel/user/Admin.java]: maybe "roomReservation" will be  more correct?

File: src/main/java/com/hotel/user/User.java (hunk 3 above)
Comment (Ivanko-Mosiychuk) [reply src/main/java/com/hotel/user/User.java]: Corrected it.

File: src/main/java/com/hotel/user/Admin.java (hunk 4 above)
Comment (Ivanko-Mosiychuk) [reply src/main/java/com/hotel/user/Admin.java]: Corrected it.

**Pull request merged**